packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityMethodChannelHandler.java

```diff
@@ -0,0 +1,163 @@
+        handleWifiIPAddress(call, result);
+        break;
+      default:
+        result.notImplemented();
+        break;
+    }
+  }
+
+  private String getNetworkType(ConnectivityManager manager) {
+    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      Network network = manager.getActiveNetwork();
+      NetworkCapabilities capabilities = manager.getNetworkCapabilities(network);
+      if (capabilities == null) {
+        return "none";
+      }
+      if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
+          || capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
+        return "wifi";
+      }
+      if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
+        return "mobile";
+      }
+    }
+
+    return getNetworkTypeLegacy(manager);
+  }
+
+  @SuppressWarnings("deprecation")
+  private String getNetworkTypeLegacy(ConnectivityManager manager) {
+    // handle type for Android versions less than Android 9
+    NetworkInfo info = manager.getActiveNetworkInfo();
+    if (info == null || !info.isConnected()) {
+      return "none";
+    }
+    int type = info.getType();
+    switch (type) {
+      case ConnectivityManager.TYPE_ETHERNET:
+      case ConnectivityManager.TYPE_WIFI:
+      case ConnectivityManager.TYPE_WIMAX:
+        return "wifi";
+      case ConnectivityManager.TYPE_MOBILE:
+      case ConnectivityManager.TYPE_MOBILE_DUN:
+      case ConnectivityManager.TYPE_MOBILE_HIPRI:
+        return "mobile";
+      default:
+        return "none";
+    }
+  }
+
+  private BroadcastReceiver createReceiver(final EventChannel.EventSink events) {
+    return new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        events.success(checkNetworkType());
+      }
+    };
+  }
+
+  private String checkNetworkType() {
+    return getNetworkType(manager);
+  }
+
+  private void handleCheck(MethodCall call, final MethodChannel.Result result) {
+    result.success(checkNetworkType());
+  }
+
+  private WifiInfo getWifiInfo() {
+    WifiManager wifiManager =
+        (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
+    return wifiManager == null ? null : wifiManager.getConnectionInfo();
+  }
+
+  private void handleWifiName(MethodCall call, final MethodChannel.Result result) {
+    WifiInfo wifiInfo = getWifiInfo();
+    String ssid = null;
+    if (wifiInfo != null) ssid = wifiInfo.getSSID();
+    if (ssid != null) ssid = ssid.replaceAll("\"", ""); // Android returns "SSID"
+    result.success(ssid);
+  }
+
+  private void handleBSSID(MethodCall call, MethodChannel.Result result) {
+    WifiInfo wifiInfo = getWifiInfo();
+    String bssid = null;
+    if (wifiInfo != null) bssid = wifiInfo.getBSSID();
+    result.success(bssid);
+  }
+
+  private void handleWifiIPAddress(MethodCall call, final MethodChannel.Result result) {
+    WifiManager wifiManager =
```

Comment:
Recommendation: passing this as a constructor argument allows you to mock for tests.